### ISSUE TITLE
Telemetry gets http.route attribute wrong when subresources are used

### ISSUE THREAD

**Issue by Azquelt** [OpenLiberty/open-liberty#25058]
In #24731, a fix was made for telemetry throwing a NPE when a resource class didn't have the `@Path` annotation on the class.

The fix ensures we don't throw an exception, but I'm not convinced we're doing the right thing.

There are a few things that need done here:

1. Find out how you can have a resource class without the `@Path` annotation in the first place. The easiest way to do this might be to revert #24731 locally and run the RestfulWS TCK and investigate the tests that fail.
2. Create a Telemetry test case for a similar scenario in the telemetry FAT project
3. Work out what the correct behaviour is in that scenario. We might need to look for a different way to get the route since at the moment we rely on finding the annotations on the class. We might need to ask the JAX-RS team for help with this.

**Comment by jakub-pomykala** [comment]
I created a application:
```Java
@ApplicationPath("/inventory")
public class InventoryApplication extends Application {
}

```

which had a resource:

```Java
@RequestScoped
@Path("/systems")
public class InventoryResource {

    @Path("{id}")
    public TestClass testing(@PathParam("id") String id) {
        return new TestClass(id);
    }
    
}

```
And a sub-resource:

```Java
public class TestClass {

    public TestClass(String id) {
        this.id = id;
    }

    private String id;
    
    @GET
    public String listContents() {
        System.out.println(id);
        return id;
    }

    @GET
    @Path("details")
    public String Details() {
        return "Details about ID";
    }
}

```

When I access `localhost:9081/inventory/systems/testing` the `http.route` returned is `/inventory` but expected was `/inventory/systems/{id}`

When I access `localhost:9081/inventory/systems/testing/details` the `http.route` returned is `/inventory/details` but expected was `/inventory/systems/{id}/details`

**Comment by Azquelt** [comment]
The JAX-RS team suggest using [UriInfo.getMatchedURIs()](https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/UriInfo.html#getMatchedURIs--)

**Comment by Azquelt** [comment]
Updated the title since we identified subresources as an issue.

**Comment by Azquelt** [comment]
We currently don't have a way to get the full route, so we're currently looking at just avoiding setting the route to something incorrect.

**Comment by Azquelt** [comment]
Fixed by #26883
